### PR TITLE
feat: add simple forward propagation implementation

### PR DESCRIPTION
**Pull request by CaioCordeiro**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### Tests Results:
```
Trying:
    res = forward_propagation(32, 10000000)
Expecting nothing
ok
Trying:
    res > 31 and res < 33
Expecting:
    True
ok
Trying:
    res = forward_propagation(32, 1000)
Expecting nothing
ok
Trying:
    res > 31 and res < 33
Expecting:
    False
ok
Trying:
    sigmoid_function(3.5)
Expecting:
    0.9706877692486436
ok
Trying:
    sigmoid_function(3.5, True)
Expecting:
    -8.75
ok
1 items had no tests:
    simple_neural_network
2 items passed all tests:
   4 tests in simple_neural_network.forward_propagation
   2 tests in simple_neural_network.sigmoid_function
6 tests in 3 items.
6 passed and 0 failed.
Test passed.
```

### CODE REVIEW

File: neural_network/simple_neural_network.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+def sigmoid_function(value: float, deriv: bool = False) -> float:
+    """Return the sigmoid function of a float.
+
+    >>> sigmoid_function(3.5)
+    0.9706877692486436
+    >>> sigmoid_function(3.5, True)
+    -8.75
+    """
+    if deriv:
+        return value * (1 - value)
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:20]:
```suggestion
           return value * (1 - value)
```

File: neural_network/simple_neural_network.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+"""
+Forward propagation explanation:
+https://towardsdatascience.com/forward-propagation-in-neural-networks-simplified-math-and-code-version-bbcfef6f9250
+"""
+
+import random as ran
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:6]: Why are you importing it as `ran`?

File: neural_network/simple_neural_network.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# Output
+expected = int(input("What is the expected value? (1-100): "))
+
+expected = expected / 100
+
+# Number of propagations
+n = int(input("How many propagations?: "))
+
+# Random weight
+syn0 = 2 * (ran.randint(1, 34)) - 1
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:36]: What does this name mean?

File: neural_network/simple_neural_network.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# Initial Value
+INITIAL_VALUE = 0.02
+
+# Output
+expected = int(input("What is the expected value? (1-100): "))
+
+expected = expected / 100
+
+# Number of propagations
+n = int(input("How many propagations?: "))
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:33]: Could this name be more descriptive?

File: neural_network/simple_neural_network.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+# Random weight
+syn0 = 2 * (ran.randint(1, 34)) - 1
+
+for _ in range(n):
+
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:39]:
```suggestion
```

File: neural_network/simple_neural_network.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    # Forward propagation
+    layer_1 = sigmoid_function((INITIAL_VALUE * syn0))
+
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:42]:
```suggestion
```

File: neural_network/simple_neural_network.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    # How much did we miss?
+    layer_1_error = expected - layer_1
+
```
Comment (CaedenPH) [inline neural_network/simple_neural_network.py:45]:
```suggestion
```

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Can you put the code into a test function and run it via some testing module?

File: neural_network/simple_neural_network.py (hunk 1 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py:20]: I don't think this is right, are you sure of this change?

File: neural_network/simple_neural_network.py (hunk 2 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py:6]: Personal preference, do you want me to change it?

File: neural_network/simple_neural_network.py (hunk 3 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: I changed the name of this variable to be more descriptive.

File: neural_network/simple_neural_network.py (hunk 4 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done

File: neural_network/simple_neural_network.py (hunk 5 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done

File: neural_network/simple_neural_network.py (hunk 6 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done.

File: neural_network/simple_neural_network.py (hunk 7 above)
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done

File: neural_network/simple_neural_network.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    # Error delta
+    layer_1_delta = layer_1_error * sigmoid_function(layer_1, True)
+
```
[the remark this answers is not in the document]
Comment (CaioCordeiro) [reply neural_network/simple_neural_network.py]: Done

File: neural_network/simple_neural_network.py (hunk 2 above)
Comment (CaedenPH) [reply neural_network/simple_neural_network.py:6]: Its up to you, but I think consistency is key and from what I can tell the algorithms just `import random` normally

File: neural_network/simple_neural_network.py (hunk 1 above)
Comment (CaedenPH) [reply neural_network/simple_neural_network.py:20]: > I don't think this is right, are you sure of this change?

It seems fixed now, perhaps it was a bug? On my screen, it seemed like there was only 1 space after the `if`

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Looks good, but one last comment from me. Can you add a way to run it at the end? Perhaps a testing module like `doctest`, or just 
```py
if __name__ = "__main__":
    ...
```

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**